wd_core.au3

```diff
@@ -1709,6 +1709,7 @@ EndFunc   ;==>__WD_StripPath
 Func __WD_ConsoleWrite($sMsg, $iDebugLevel = Default, $iError = @error, $iExtended = @extended)
 	If $iDebugLevel = Default Or $_WD_DEBUG >= $iDebugLevel Then
 		If IsFunc($_WD_CONSOLE) Then
+			If IsNumber($iDebugLevel) And $iDebugLevel = $_WD_DEBUG_Full Then $sMsg = ':$_WD_DEBUG_Full: (' & $iError & '/' & $iExtended & ') : ' & $sMsg
```

Comment:
Let's talk this one through. Please respond to each of the following items --

- Is the `IsNumber` check required? What happens without it?
- Why doesn't this extended logging apply to file logs?
- How could this negatively effect the scenario where the target is a user-supplied function?
- Show an example of the output for both scenarios (`$iDebugLevel =  $_WD_DEBUG_Full` and `$iDebugLevel <>  $_WD_DEBUG_Full`

Reply:
> * Is the `IsNumber` check required? What happens without it?

Take a look on: [small fix](https://github.com/Danp2/au3WebDriver/pull/291/commits/e1028d3fad7601e3dcc80136092ee950fc17af42)

> * Why doesn't this extended logging apply to file logs?

Take a look on: [small fix](https://github.com/Danp2/au3WebDriver/pull/291/commits/e1028d3fad7601e3dcc80136092ee950fc17af42)

> * How could this negatively effect the scenario where the target is a user-supplied function?

As so far `$_WD_DEBUG_Full` is not in officiall release.
`$_WD_DEBUG_Full` is not commonly used.
This is user responsibility to support `$_WD_DEBUG_Full` messages in a `user-supplied function` 

> * Show an example of the output for both scenarios (`$iDebugLevel =  $_WD_DEBUG_Full` and `$iDebugLevel <>  $_WD_DEBUG_Full`

give me some time...

Reply:
> Small fix

That looks better, although It's a pretty big fix if you ask me. 😜 

> As so far $_WD_DEBUG_Full is not in officiall release

Right, but _WD_ConsoleWtire() to a custom function is official and could be in use by others. Thus, you should consider how your proposal could affect existing code.

Reply:
We should add a remark to release.
`in the case when the new type was used  $_WD_DEBUG_Full the content of the message sent to the console is preceded by ':$_WD_DEBUG_Full: (' & $iError & '/' & $iExtended & ') : '`

And this is not SCRIPT BREAKING CHANGE as far as `$_WD_DEBUG_Full` is not used, which is very uncommon to use them.

Reply:
We can change:
```autoit
':$_WD_DEBUG_Full: (' & $iError & '/' & $iExtended & ') : '
``
to
```autoit
'[' & $iError & '/' & $iExtended & '] $_WD_DEBUG_Full: '
``
Which should be easier to to manage the change in `user-supplied function` becuase it will require only simple RegExp to delete this additionall message but this is still required to have `$iError' and '$iExtended' in log and have `$_WD_DEBUG_Full` in prefix to the message because, it is DEVELOPER messages so you must be able to recognize in the whole LOG file that this is a development message and at the same time you must have the previous function execution status to be able to diagnose a given test scenario.

Reply:
I suspect that we have different ideas on how `$_WD_DEBUG` should be implemented. It seems like you want to change the logging **content** based on the logging level. This isn't in line with my vision of how the logging levels should work. 

If we leave in the `$_WD_DEBUG_Full` portion, then it should be user configurable, similar to the $_WD_CONSOLE_Suffix setting. This would allow the script writer to change the value or eliminate it all together.

Additionally, we could adjust some of the existing logging so that it only occurs when `$_WD_DEBUG = $_WD_DEBUG_Full`. For example, __WD_Get, __WD_Post, and __WD_Delete could be modified so that their logging only occurs when `$_WD_DEBUG = $_WD_DEBUG_Full`. I believe this would address some of your issues listed in #290

Reply:
> I suspect that we have different ideas on how `$_WD_DEBUG` should be implemented. It seems like you want to change the logging **content** based on the logging level. This isn't in line with my vision of how the logging levels should work.

I just change the concept a little.
```autoit
If $iDebugLevel = $_WD_DEBUG_Full Then $sMsg &= ' [' & $iError & '/' & $iExtended & ']'
```
instead:
```autoit
If $iDebugLevel = $_WD_DEBUG_Full Then $sMsg = ':$_WD_DEBUG_Full: (' & $iError & '/' & $iExtended & ') : ' & $sMsg
```
Now when somebody INTENTIONALLY use `$_WD_DEBUG_Full` in such way:
```autoit
__WD_ConsoleWrite(' Test #' & @ScriptLineNumber, $_WD_DEBUG_Full)
```

then the only one difference will be `' [' & $iError & '/' & $iExtended & ']'` added at the end of message, which should be any problem for any `user-supplied function`

I must also clearly indicate that this modification it will not affect message when `__WD_Error()` is used
```autoit
	Return SetError(__WD_Error($sFuncName, $_WD_ERROR_Exception), 0, "")
```

And it will not affect message in such cases:
```autoit
	__WD_ConsoleWrite($sFuncName & ": " & $sResponse, $_WD_DEBUG_Info)
```

> If we leave in the `$_WD_DEBUG_Full` portion, then it should be user configurable, similar to the $_WD_CONSOLE_Suffix setting.


I must again clearly indicate that this modification it will not affect message when `__WD_Error()` is used, and will not affect:
```autoit
	__WD_ConsoleWrite($sFuncName & ": " & $sResponse, $_WD_DEBUG_Info)
```
It affect only specific message:
```autoit
__WD_ConsoleWrite(' Test #' & @ScriptLineNumber, $_WD_DEBUG_Full)
```

> This would allow the script writer to change the value or eliminate it all together.

Currently it is very easy I will prepare example soon.


> Additionally, we could adjust some of the existing logging so that it only occurs when `$_WD_DEBUG = $_WD_DEBUG_Full`. For example, __WD_Get, __WD_Post, and __WD_Delete could be modified so that their logging only occurs when `$_WD_DEBUG = $_WD_DEBUG_Full`. I believe this would address some of your issues listed in #290

We can discuss this in https://github.com/Danp2/au3WebDriver/pull/290
But firstly I want to end my proposals in #290 currently this is still draft and I have some progress done in the background.

You must wait a little more with #290

Reply:
take a look on:

```autoit
#include "wd_core.au3"

_Example()
Func _Example()
	Local $sSesion

	ConsoleWrite("> Testing $_WD_DEBUG_None" & @CRLF)
	$_WD_DEBUG = $_WD_DEBUG_None
	_WD_DeleteSession($sSesion)
	__WD_ConsoleWrite(': Test #' & @ScriptLineNumber, $_WD_DEBUG_Full)
	ConsoleWrite(@CRLF)
	ConsoleWrite(@CRLF)

	ConsoleWrite("> Testing $_WD_DEBUG_Error" & @CRLF)
	$_WD_DEBUG = $_WD_DEBUG_Error
	_WD_DeleteSession($sSesion)
	__WD_ConsoleWrite(': Test #' & @ScriptLineNumber, $_WD_DEBUG_Full)
	ConsoleWrite(@CRLF)
	ConsoleWrite(@CRLF)

	ConsoleWrite("> Testing $_WD_DEBUG_Info" & @CRLF)
	$_WD_DEBUG = $_WD_DEBUG_Info
	_WD_DeleteSession($sSesion)
	__WD_ConsoleWrite(': Test #' & @ScriptLineNumber, $_WD_DEBUG_Full)
	ConsoleWrite(@CRLF)
	ConsoleWrite(@CRLF)

	ConsoleWrite("> Testing $_WD_DEBUG_Full" & @CRLF)
	$_WD_DEBUG = $_WD_DEBUG_Full
	_WD_DeleteSession($sSesion)
	__WD_ConsoleWrite(': Test #' & @ScriptLineNumber, $_WD_DEBUG_Full)
	ConsoleWrite(@CRLF)
	ConsoleWrite(@CRLF)

EndFunc   ;==>_Example

```

![image](https://user-images.githubusercontent.com/11089482/163424497-bfc7ff17-6fa2-4a99-ad11-1ed56188c2b1.png)


as you can see I used
```autoit
	__WD_ConsoleWrite(': Test #' & @ScriptLineNumber, $_WD_DEBUG_Full)
```
multiple time and only on last was resulted in message in console.

Also only this last call have added sufix ` [10/0]`

Is this fine for you ?